internal/provider/resource_cloudflare_teams_list.go

```diff
@@ -70,14 +70,32 @@ func resourceCloudflareTeamsListRead(ctx context.Context, d *schema.ResourceData
 	d.Set("type", list.Type)
 	d.Set("description", list.Description)
 
-	listItems, _, err := client.TeamsListItems(ctx, cloudflare.TeamsListItemsParams{
-		AccountID: accountID,
-		ListID:    d.Id(),
+	var allListItems []cloudflare.TeamsListItem
+
+	listItems, responseInfo, err := client.TeamsListItems(ctx, cloudflare.TeamsListItemsParams{
+		AccountID:         accountID,
+		ListID:            d.Id(),
+		PaginationOptions: cloudflare.PaginationOptions{PerPage: 1000},
 	})
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("error finding Teams List %q: %w", d.Id(), err))
 	}
-	d.Set("items", convertListItemsToSchema(listItems))
+
+	allListItems = append(allListItems, listItems...)
+
+	for i := 2; i < responseInfo.TotalPages; i++ {
```

Comment:
Shouldn't this be `<=`?

Reply:
Actually, in my list that has 1240 items, if I query the API for page 2 with a page size of 10 I get the following response:
```
  "result_info": {
    "page": 2,
    "per_page": 10,
    "count": 10,
    "total_count": 1240
  }
```
There is no TotalPages in there.   Is that something the cloudflare-go library is determining?

Reply:
If i query for page 2 with a page size of 1000, I get:
```
  "result_info": {
    "page": 2,
    "per_page": 1000,
    "count": 240,
    "total_count": 1240
  }
```
So maybe it should be checking if the `count` returned is less than the `page_size` requested to see if there's more?  Or, I guess do math to figure out if it got the last item on the list with something like `(($page-1) * $per_page) + $count) < $total_count`.  The API response is kinda terrible.  Normally there's a clear indicator if there are additional items and a `next` token or page number to key off of.

Reply:
`TotalPages` is a part of the v4 envelope that teams should be emitting for paginated results. let me confirm why it isn't present here before we swap to using the total count instead.